src/main.rs

```diff
@@ -34,16 +55,122 @@ async fn capacity(pool: &State<db::AppDbPool>) -> Result<Json<Capacity>> {
     Ok(Json(db::network_capacity(&pool.mongo).await?))
 }
 
+#[post("/stats", format = "application/json", data = "<stats>")]
+async fn update_stats(stats: HostStats, pool: &State<db::AppDbPool>) -> Result<(), ApiError> {
+    let ed25519_pubkey = decode_pubkey(&stats.holoport_id_base36);
+
+    // Confirm host exists in registration records
+    let _ = db::verify_host(
+        stats.email.clone(),
+        to_holochain_encoded_agent_key(&ed25519_pubkey),
+        &pool.mongo,
+    )
+    .await
+    .or_else(|e| {
+        return Err(ApiError::MissingRecord(ErrorMessageInfo(format!(
+            "Provided host's holoport_id is not registered among valid hosts.  Error: {:?}",
+            e
+        ))));
+    });
+
+    // Add utc timestamp to stats payload and insert into db
+    let holoport_status = HostStats {
+        holo_network: stats.holo_network,
+        channel: stats.channel,
+        holoport_model: stats.holoport_model,
+        ssh_status: stats.ssh_status,
+        zt_ip: stats.zt_ip,
+        wan_ip: stats.wan_ip,
+        holoport_id_base36: stats.holoport_id_base36,
+        timestamp: format!("{:?}", SystemTime::now()),
+        email: stats.email,
+    };
+    db::add_holoport_status(holoport_status, &pool.mongo).await?;
+    Ok(())
```

Comment:
If we move all this code into a function `update_stats` in `db` module we are gonna have much cleaner `main.rs`. The idea behind it is that in `main.rs` you create endpoints and all the handlers are in separate module.